lua/ibl/utils.lua

```diff
@@ -9,11 +9,25 @@ M.get_whitespace = function(line)
     return string.match(line, "^%s+") or ""
 end
 
+--- Use the faster validate version if available.
+---@param spec table<string, {[1]:any, [2]:function|string, [3]:string|true|nil}>
+M.validate = function(spec)
+    if vim.fn.has "nvim-0.11" == 1 then
+        for key, key_spec in pairs(spec) do
+            local message = type(key_spec[3]) == "string" and key_spec[3] or nil
+            local optional = type(key_spec[3]) == "boolean" and key_spec[3] or nil
```

Comment:
lua-ls isn't smart enough to do this type narrowing automatically, `message` and `optional` are still `string|true|nil` and CI complains. You can just cast it here.

```suggestion
            local message = type(key_spec[3]) == "string" and key_spec[3] or nil --[[ @as string? ]]
            local optional = type(key_spec[3]) == "boolean" and key_spec[3] or nil --[[ @as boolean? ]]
```